src/Dhall/Parser.hs

```diff
@@ -101,26 +91,34 @@ instance Monoid a => Monoid (Parser a) where
     mappend = liftA2 mappend
 
 instance IsString a => IsString (Parser a) where
-    fromString x = fmap fromString (Text.Parser.Char.string x)
+    fromString x = fromString x <$ Text.Megaparsec.Char.string (fromString x)
+
+instance Text.Parser.Combinators.Parsing Parser where
+  unexpected = fail
+  eof = Parser Text.Megaparsec.eof
+  try = Text.Megaparsec.try
+  (<?>) = (Text.Megaparsec.<?>)
+  notFollowedBy = Text.Megaparsec.notFollowedBy
+
+instance Text.Parser.Char.CharParsing Parser where
+  satisfy = Parser . Text.Megaparsec.Char.satisfy
 
 instance TokenParsing Parser where
     someSpace =
         Text.Parser.Token.Style.buildSomeSpaceParser
-            (Parser someSpace)
+            (Parser (Text.Megaparsec.skipSome (Text.Megaparsec.Char.satisfy Data.Char.isSpace)))
```

Comment:
Do you thing replacing `satisfy isSpace` with something ala `sc = L.space space1 lineComment blockComment` might make a difference performance wise ?

Reply:
No idea! You're welcome to grab my branch and see :) I won't be able to check that for a while, probably not free to hack now until Sunday.